Network/DNS/Transport.hs

```diff
@@ -23,8 +23,17 @@ import Network.DNS.Types.Internal
 -- pipelined TCP, we'll need to handle out of order responses.  See:
 -- https://tools.ietf.org/html/rfc7766#section-7
 checkResp :: Question -> Identifier -> DNSMessage -> Bool
-checkResp q seqno resp =
-   identifier (header resp) == seqno && [q] == question resp
+checkResp q seqno = isNothing . checkRespM (Just q) seqno
+
+checkRespM :: Maybe Question -> Identifier -> DNSMessage -> Maybe DNSError
+checkRespM qM seqno resp =
+    if identifier (header resp) /= seqno
+    then Just SequenceNumberMismatch
+    else do
+      qs <- (:[]) <$> qM
+      if qs /= question resp
+      then Just QuestionMismatch
+      else Nothing
```

Comment:
The new function should not take a question, no reason for `Maybe` arguments.

Reply:
The main purpose of this new function is to differentiate between sequence number and question mismatch. I can make it take a `Question` instead of a `Maybe Question.